planner/core/planbuilder.go

```diff
@@ -1028,6 +1039,10 @@ func getPossibleAccessPaths(ctx sessionctx.Context, tableHints *tableHintInfo, i
 				ctx.GetSessionVars().StmtCtx.AppendWarning(err)
 				continue
 			}
+			if isGlobalIndex(idxName, tblInfo) {
```

Comment:
The path got by the `idxName` and `tblInfo` of line#1032 already contains the `*model.IndexInfo`. So you can directly use it instead of a new function I think.